src/webui/webapplication.cpp

```diff
@@ -940,6 +940,7 @@ void WebSession::registerAPIController(const QString &scope, APIController *cont
 {
     Q_ASSERT(controller);
     m_apiControllers[scope] = controller;
+    controller->setParent(this);
```

Comment:
On a closer look this is not the most direct way to solve it.

I would suggest the fix to be applied here:
https://github.com/qbittorrent/qBittorrent/blob/8e941a06f15a0a8c12f37439cb2815fec73797c3/src/webui/webapplication.cpp#L733-L744

And provide the correct `parent`. For example, change to `new AppController(app(), m_currentSession)`, `new LogController(app(), m_currentSession)` and so on.

Reply:
The session registers the controller and becomes its owner. Isn't that logical?
I don't mind either way.

Reply:
In this case I would also vote on setting the parent in the `APIController`'s ctor. Right now it is set first to the `WebApplication`, then almost immediately overwritten with a new parent from the session's ctor.
I'll test it for a short period of time then push the changes if works fine (should be).